cmd/operator-sdk/alpha/bundle/validate.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// newBundleBuildCmd returns a command that will build operator bundle image.
+func newBundleValidateCmd() *cobra.Command {
+	c := bundleCmd{}
+	cmd := &cobra.Command{
+		Use:   "validate",
+		Short: "Validate an operator bundle image",
+		Long: `The 'operator-sdk bundle validate' command will validate both content and
+format of an operator bundle image containing operator metadata and manifests.
+This command will exit with a non-zero exit code if any validation tests fail.`,
+		Example: `The following invocation will validate a test-operator bundle image hosted
+in a Docker image registry:
+
+$ operator-sdk bundle validate quay.io/example/operator:v0.1.0 --image-builder docker`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("a bundle image tag is a required argument, ex. example.com/test-operator:v0.1.0")
+			}
+			c.imageTag = args[0]
+
+			dir, err := ioutil.TempDir("", "bundle-")
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer func() {
+				if err = os.RemoveAll(dir); err != nil {
+					log.Error(err.Error())
+				}
+			}()
+			logger := log.WithFields(log.Fields{"container-tool": c.imageBuilder})
+			log.SetLevel(log.DebugLevel)
+			val := bundle.NewImageValidator(c.imageBuilder, logger)
+			if err = val.PullBundleImage(c.imageTag, dir); err != nil {
```

Comment:
Does this command work if I have a bundle image on my local system and I haven't pushed it to a registry yet?

Reply:
Apparently not. I tested this command with a few images: local ones all returning the same error, while registry images were fine. I'll add a note here.

Additionally error messages can be improved in `PullBundleImage`, as right now the only message I'm seeing is `exit status 1` @exdx @dinhxuanvu.